apps/web/src/components/workflow/edge-types/AddNodeEdge.tsx

```diff
@@ -0,0 +1,86 @@
+import { EdgeProps, getBezierEdgeCenter, getBezierPath } from 'react-flow-renderer';
+import AddNode from '../node-types/AddNode';
+import styled from '@emotion/styled';
+
+const foreignObjectSize = 30;
+
+export function AddNodeEdge({
+  id,
+  sourceX,
+  sourceY,
+  targetX,
+  targetY,
+  sourcePosition,
+  targetPosition,
+  style = {},
+  data,
+  markerEnd,
+}: EdgeProps) {
+  const [centerX, centerY] = getBezierEdgeCenter({
+    sourceX,
+    sourceY,
+    sourcePosition,
+    targetX,
+    targetY,
+    targetPosition,
+  });
+
+  const edgePath = getBezierPath({
+    sourceX,
+    sourceY,
+    sourcePosition,
+    targetX,
+    targetY,
+    targetPosition,
+  });
+
+  const { childId, parentId, addNewNode } = data;
+
+  return (
+    <>
+      <path style={style} className="react-flow__edge-path" d={edgePath} markerEnd={markerEnd} />
+
+      <NodeWrapper
+        width={foreignObjectSize}
+        height={foreignObjectSize}
+        x={centerX - foreignObjectSize / 2}
+        y={centerY - foreignObjectSize / 2}
+        requiredExtensions="http://www.w3.org/1999/xhtml"
+        id={`edge-button-${parentId}`}
+        className={id}
+      >
+        <div>
+          <AddNode
+            data={{
+              parentId: parentId,
+              addNewNode: addNewNode,
+              showDropZone: false,
+              childId: childId,
+            }}
+          />
+        </div>
+      </NodeWrapper>
+    </>
+  );
+}
+
+const NodeWrapper = styled.foreignObject`
+  opacity: 0;
+  height: 30px;
+  width: 30px;
```

Comment:
I understand these values depend on the assigned `foreignObjectSize` value. We can use the template literal to have a single source of truth.
```suggestion
  height: ${foreignObjectSize}px;
  width: ${foreignObjectSize}px;
```

Reply:
Good catch! I totally agree thanks I'll update it :)